tla/consensus/ccfraft.tla

```diff
@@ -456,28 +456,49 @@ PlausibleSucessorNodes(i) ==
+    /\ currentTerm = [i \in servers |-> IF i = startNode THEN 2 ELSE 0]
+    /\ state       = [i \in servers |-> IF i = startNode THEN Leader ELSE None]
+    /\ votedFor    = [i \in servers |-> Nil]
+    /\ log         = [i \in servers |-> IF i = startNode
+                                        THEN << [term |-> 2, contentType |-> TypeReconfiguration, configuration |-> {startNode}],
+                                                [term |-> 2, contentType |-> TypeSignature] >>
+                                        ELSE << >>]
+    /\ commitIndex  = [i \in servers |-> IF i = startNode THEN 2 ELSE 0]
+    /\ committableIndices  = [i \in servers |-> {}]
+
+\* Generate initial state for a startNodes configuration
+JoinedState(startNodes, servers) ==
+    /\ reconfigurationCount = 0
+    /\ removedFromConfiguration = {}
+    /\ \E startNode \in startNodes:
+        /\ configurations = [ i \in servers |-> IF i \in startNodes  THEN (3 :> startNodes) ELSE << >>]
+        /\ currentTerm = [i \in servers |-> IF i \in startNodes THEN 2 ELSE 0]
+        /\ state       = [i \in servers |-> IF i = startNode THEN Leader ELSE IF i \in startNodes THEN Follower ELSE None]
+        /\ votedFor    = [i \in servers |-> Nil]
+        /\ log         = [i \in servers |-> IF i \in startNodes
+                                            THEN << [term |-> 2, contentType |-> TypeReconfiguration, configuration |-> {startNode}],
+                                                    [term |-> 2, contentType |-> TypeSignature],
+                                                    [term |-> 2, contentType |-> TypeReconfiguration, configuration |-> startNodes],
+                                                    [term |-> 2, contentType |-> TypeSignature] >>
+                                            ELSE << >>]
+        /\ commitIndex  = [i \in servers |-> IF i \in startNodes THEN 4 ELSE 0]
+        /\ committableIndices  = [i \in servers |-> {}]
+
 ------------------------------------------------------------------------------
 \* Define initial values for all variables
 
 InitReconfigurationVars ==
-    /\ reconfigurationCount = 0
-    /\ removedFromConfiguration = {}
-    \* Note that CCF has a bootstrapping procedure to start a new network and to join new nodes to the network (see 
-    \* https://microsoft.github.io/CCF/main/operations/start_network.html). In both cases, a node has the current (see 
-    \* https://microsoft.github.io/CCF/main/operations/ledger_snapshot.html#join-or-recover-from-snapshot) or some stale configuration
-    \* such as the initial configuration. A node's configuration is *never* "empty", i.e., the equivalent of configuration[node] = {} here. 
-    \* For simplicity, the set of servers/nodes all have the same initial configuration at startup.
-    /\ \E s \in Servers:
-        configurations = [i \in Servers |-> 0 :> {s} ]
+    \E startNode \in Servers:
```

Comment:
Quantification of `startNodes` should fail in `StartState` and `JoinedState` because `startNode` cannot be quantified over.  `SUBSET Servers` maybe?